### PR TITLE
#184 Search Page Simplifications

### PR DESCRIPTION
**Pull request by xXCrash2BomberXx**
Instead of having a bar to create a Label only gloss and another bar for searching, these bars should be unified allowing the user to click to create a gloss if they can't find one by searching instead of having to copy over the text contents between bars
Old:
image

New:
image

### CODE REVIEW

**Review by thehabes — CHANGES_REQUESTED** [review]
Some immediate thingsI noticed...

First, you are allowed to come into the page with a search query.  When I do, the area does not make sense.

glosses.html?gog-filter=JTdCJTIydGl0bGUlMjIlM0ElMjJjb25jZXJuJTIyJTJDJTIydGV4dCUyMiUzQSUyMmNvbmNlcm4lMjIlMkMlMjJ0YXJnZXRlZHRleHQlMjIlM0ElMjJjb25jZXJuJTIyJTdE
image

Second, there are no instructions for what "New Gloss" does.  Add some instructions.
image

Third, when I click "New Gloss", it submits an empty, untitled Gloss.  That button should not actually create an entity, especially since there are no instructions telling you what you should expect to have happen.  You should use some kind of confirm modal to have the user confirm they want to make a Gloss with the title they typed.

Fourth, you have resurrected a bug.  When I use the "Enter" button on my keyboard while my cursor is in the filter it submits the form and creates the Gloss.  "Enter" should not cause that form to submit and right now does not at https://glossing.rerum.io

Take your time and be a little more careful here.  This should stay in Draft until you have done more testing.

File: js/deer-config.js [hunk 1]
```diff
@@ -161,7 +161,12 @@ export default {
                 </style>
                 <h2> Glosses </h2>
                 <small class="cachedNotice is-hidden text-primary"> These Glosses were cached.  To reload the data <a class="newcache tag is-small">click here</a>. </small>
-                <input filter="title" type="text" placeholder="&hellip;Type to filter by incipit, text, or targeted text" class="is-hidden serifText">
+				<form id="ngForm" deer-type="named-gloss" deer-context="http://www.loc.gov/mods" class="row">
```
Comment (thehabes) [inline js/deer-config.js:164]: The value for the deer-type attribute should be "Gloss".  "named-gloss" is the old type and we have since changed it.

**Review by thehabes — CHANGES_REQUESTED** [review]
Much better, it worked nicely this time.  There a couple of small things left.

The message that appears when making the new Gloss is just "Saving Successful".  I think it may have said "Gloss Saved" but the text was quickly replaced.  The message should let the user know that a Gloss was created and they are being redirected to that Gloss detail page.  Here's a screenshot.

image

Make sure not to use deer-type="named-gloss" anymore, that's the old name.  I did not realize any of those were still around.  It is ok for a form to have `id="named-gloss"` but not `deer-type="named-gloss"`.  You'll see that comment inline with the code where I noticed it.

File: js/deer-config.js [hunk 2]
```diff
@@ -161,7 +161,12 @@ export default {
                 </style>
                 <h2> Glosses </h2>
                 <small class="cachedNotice is-hidden text-primary"> These Glosses were cached.  To reload the data <a class="newcache tag is-small">click here</a>. </small>
-                <input filter="title" type="text" placeholder="&hellip;Type to filter by incipit, text, or targeted text" class="is-hidden serifText">
+				<form id="ngForm" deer-type="Gloss" deer-context="http://www.loc.gov/mods" class="row">
```
Comment (cubap) [inline js/deer-config.js:164]: I'm not certain why this is a form. There isn't a need for a creator or anything that I can see, just processing the search term. Let me know if I'm missing something.

File: glosses.html [hunk 3]
```diff
@@ -48,10 +42,13 @@
...
-            let e = new CustomEvent("deer-view", { detail: { set: [ngCollectionList] }, bubbles: true })
-            document.dispatchEvent(e)
-            setTimeout(() => location.href = "", 3000)
+        document.addEventListener('deer-updated', event => {
+            globalFeedbackBlip(
+				new CustomEvent("deer-view", { detail: { set: [ngCollectionList] }, bubbles: true }),
+				'Gloss Created. Redirecting to Details...',
+				true
+			)
+			addEventListener('globalFeedbackFinished', () => location.href = `ng.html#${event.detail['@id']}`)
```
Comment (cubap) [inline glosses.html:51]: This looks like a better way to manage it - good.

File: js/deer-config.js [hunk 4]
```diff
@@ -291,6 +296,17 @@ export default {
                         if(filterPresent){
                             debounce(filterGlosses(elem.$contentState))
                         }
+						const searchSubmit = document.getElementById('search-submit')
```
Comment (cubap) [inline js/deer-config.js:299]: The whitespace looks weird here...

File: js/deer-config.js [hunk 5]
```diff
@@ -752,6 +779,17 @@ export default {
                         })
                         containingListElem.setAttribute("ng-list-loaded", "true")
                         deerUtils.broadcast(undefined, "ng-list-loaded", containingListElem, {})
+						const searchSubmit = document.getElementById('search-submit')
```
Comment (cubap) [inline js/deer-config.js:782]: If this is just copy-paste from line 613, it shoudl be pulled out into a function and reused.

File: js/deer-config.js [hunk 6]
```diff
@@ -886,6 +924,17 @@ export default {
...
```
Comment (cubap) [inline js/deer-config.js:927]: If this is just copy-paste from line 613, it should be pulled out into a function and reused.

**Review by cubap — CHANGES_REQUESTED** [review]
I haven't fully tested it yet, but there are some changes to make for now or comments to respond to.

**Review by AbhilashKotha — COMMENTED** [review]
Works well for me. The design is cleaner than how it was done previously.

**Review by thehabes — APPROVED** [review]
Works good and I like how the design ended up.  I still feel like there should be a confirm action after clicking the "can't find what you're looking for?" text because it is so freaking tempting to click it even if you aren't planning to.

**Pull request merged**